### PR TITLE
docs: improve contibuting documentation

### LINKED ISSUE #393: Update contributing guide
Looking at [buildkit](https://github.com/moby/buildkit/blob/master/.github/CONTRIBUTING.md) as an inspiration, we need a lot more to help guide contributors.

### PR DESCRIPTION
**Pull request by TimJones**
Signed-off-by: Tim Jones <timniverse@gmail.com>

fixes #393

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,7 +30,47 @@ conform enforce --commit-msg-file $1
 EOF
 ```
 
+In addition, all commits should be signed by the commiter.
```
Comment (bradbeam) [inline CONTRIBUTING.md]: We should probably add `git commit -s` here.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,7 +30,47 @@ conform enforce --commit-msg-file $1
...
+
 ### Pull Requests
 
-To avoid multiples CI runs, please ensure that you are running a full build before submitting your PR.
+To avoid multiples CI runs, please ensure that you are running a full build before submitting your PR, and
+branches should be squashed to a single commit.
+
+## Developing
+
+To start developing for Talos yo will need at least `GNU Make` and `Golang` v1.11.
```
Comment (andrewrynhard) [inline CONTRIBUTING.md]:
```suggestion
To start developing for Talos you will need at least `GNU Make` and `Golang` v1.11.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (andrewrynhard) [reply CONTRIBUTING.md]: Agree. Perhaps even spell out that this is different than GPG signing. That has already confused a number of people.

**Comment by TimJones** [comment]
@andrewrynhard @bradbeam I made the changes suggested if you want to review again.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -30,7 +30,49 @@ conform enforce --commit-msg-file $1
 EOF
 ```
 
+In addition, all commits should be signed by the committer using `git commit -s` which should produce a commit
+message with `Signed-off-by: Your Name <your@email>`. It in not necessary to cryptographically sign commits
```
Comment (andrewrynhard) [inline CONTRIBUTING.md]:
```suggestion
message with `Signed-off-by: Your Name <your@email>`. It is not necessary to cryptographically sign commits
```

**Comment by TimJones** [comment]
English needs a better linter ;)

**Pull request merged**